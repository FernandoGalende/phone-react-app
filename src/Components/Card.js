import React, { Component } from 'react';
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

export class Card extends Component {
  static propTypes = {
    id: PropTypes.string,
    title: PropTypes.string,
    color: PropTypes.string,
    image: PropTypes.string
  }

  render(){
    const { id, title, image, color } = this.props
    return (
      <Link to={`/detail/${id}`} className="card">
        <div className="card-image">
          <figure className="image">
            <img
              src={image}
              alt={title}
            />
          </figure>
        </div>
        <div className="card-content">
          <div className="media">
            <div className="media-content">
              <p className="title is-4">{ title }</p>
              <p className="subtitle is-6">{ color }</p>
            </div>
          </div>
        </div>
      </Link>
    )
  }
}
