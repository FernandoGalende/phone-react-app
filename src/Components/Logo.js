import React, { Component } from 'react';
import logo from '../assets/logo.svg';

class Logo extends Component {
  render() {
    return (
      <div>
        <img
          src={logo}
          className="App-logo"
          alt="logo"
        />
    </div>
    )
  }
}

export default Logo
