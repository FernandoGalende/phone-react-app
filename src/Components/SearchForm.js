import React, { Component } from 'react';

const BASE_URL=process.env.REACT_APP_BASEURL

export class SearchForm extends Component {
  options = {
    method: 'GET',
    headers: { 'Access-Control-Allow-Origin': '*'}
  }

  _handleChange = (e) => {
    this.setState({inputMovie: e.target.value})
  }

  _handleSubmit = (e) => {
    e.preventDefault()

    fetch(`${BASE_URL}/phones`,this.options)
    .then( res => res.json() )
    .then( results => {
       this.props.onResults( results.data )
    })
  }

  render () {
    return(
      <form onSubmit={ this._handleSubmit } id="search-form">
        <div className="field has-addons">
          <div className="control">
            <input
              className="input"
              onChange={ this._handleChange }
              type="text"
              placeholder="Find a phone"/>
          </div>
          <div className="control">
            <button id="search-button"
              className="button is-info">
              Search
            </button>
          </div>
        </div>
      </form>
    )
  }
}
