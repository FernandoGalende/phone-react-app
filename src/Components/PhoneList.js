import React, { Component } from 'react';
import PropTypes from 'prop-types'

import { Card } from '../Components/Card'

export class PhoneList extends Component {
  static propTypes = {
    phones: PropTypes.array,
  }

  render(){
    const { phones } = this.props
    return(
      <div className="PhonesList">

      {
        phones.map(phone => {
          return (
            <div
              key={ phone.id }
              className="PhonesList-Item">
                <Card
                  id = { phone.id }
                  title = { phone.title }
                  color = { phone.color }
                  image = { phone.thumbnails }
                />
            </div>
          )
        })
      }
      </div>
    )
  }
}