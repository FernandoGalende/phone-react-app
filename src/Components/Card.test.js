import React from 'react';
import { shallow } from 'enzyme'

//Components
import { Card } from '../Components/Card'

describe('Card component', () => {
  it('should render Card component', () => {
    const cardComponent = shallow(<Card debug/>)
    expect(cardComponent).toMatchSnapshot()
  });

  it('should render Card component with properly elements', () => {
    const id = "01"
    const title = "iPhone XS Max"
    const color = "Gold"
    const image = "https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"

    const cardComponent = shallow(<Card
        id = { id }
        title = { title }
        color = { color }
        image = { image }
      />
    )

    expect(cardComponent).toMatchSnapshot()
  });
})
