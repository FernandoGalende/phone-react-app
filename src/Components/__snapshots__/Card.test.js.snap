// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render Card component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="card-image"
        >
          <figure
            className="image"
          >
            <img />
          </figure>
        </div>,
        <div
          className="card-content"
        >
          <div
            className="media"
          >
            <div
              className="media-content"
            >
              <p
                className="title is-4"
              />
              <p
                className="subtitle is-6"
              />
            </div>
          </div>
        </div>,
      ],
      "className": "card",
      "to": "/detail/undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <figure
            className="image"
          >
            <img />
          </figure>,
          "className": "card-image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img />,
            "className": "image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": undefined,
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "figure",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="media"
          >
            <div
              className="media-content"
            >
              <p
                className="title is-4"
              />
              <p
                className="subtitle is-6"
              />
            </div>
          </div>,
          "className": "card-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="media-content"
            >
              <p
                className="title is-4"
              />
              <p
                className="subtitle is-6"
              />
            </div>,
            "className": "media",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="title is-4"
                />,
                <p
                  className="subtitle is-6"
                />,
              ],
              "className": "media-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "title is-4",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "subtitle is-6",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="card-image"
          >
            <figure
              className="image"
            >
              <img />
            </figure>
          </div>,
          <div
            className="card-content"
          >
            <div
              className="media"
            >
              <div
                className="media-content"
              >
                <p
                  className="title is-4"
                />
                <p
                  className="subtitle is-6"
                />
              </div>
            </div>
          </div>,
        ],
        "className": "card",
        "to": "/detail/undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <figure
              className="image"
            >
              <img />
            </figure>,
            "className": "card-image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img />,
              "className": "image",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": undefined,
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "figure",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="media"
            >
              <div
                className="media-content"
              >
                <p
                  className="title is-4"
                />
                <p
                  className="subtitle is-6"
                />
              </div>
            </div>,
            "className": "card-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="media-content"
              >
                <p
                  className="title is-4"
                />
                <p
                  className="subtitle is-6"
                />
              </div>,
              "className": "media",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="title is-4"
                  />,
                  <p
                    className="subtitle is-6"
                  />,
                ],
                "className": "media-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "title is-4",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "subtitle is-6",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Card component should render Card component with properly elements 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    color="Gold"
    id="01"
    image="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
    title="iPhone XS Max"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="card-image"
        >
          <figure
            className="image"
          >
            <img
              alt="iPhone XS Max"
              src="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
            />
          </figure>
        </div>,
        <div
          className="card-content"
        >
          <div
            className="media"
          >
            <div
              className="media-content"
            >
              <p
                className="title is-4"
              >
                iPhone XS Max
              </p>
              <p
                className="subtitle is-6"
              >
                Gold
              </p>
            </div>
          </div>
        </div>,
      ],
      "className": "card",
      "to": "/detail/01",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <figure
            className="image"
          >
            <img
              alt="iPhone XS Max"
              src="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
            />
          </figure>,
          "className": "card-image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="iPhone XS Max"
              src="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
            />,
            "className": "image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "iPhone XS Max",
              "src": "https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "figure",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="media"
          >
            <div
              className="media-content"
            >
              <p
                className="title is-4"
              >
                iPhone XS Max
              </p>
              <p
                className="subtitle is-6"
              >
                Gold
              </p>
            </div>
          </div>,
          "className": "card-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="media-content"
            >
              <p
                className="title is-4"
              >
                iPhone XS Max
              </p>
              <p
                className="subtitle is-6"
              >
                Gold
              </p>
            </div>,
            "className": "media",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="title is-4"
                >
                  iPhone XS Max
                </p>,
                <p
                  className="subtitle is-6"
                >
                  Gold
                </p>,
              ],
              "className": "media-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "iPhone XS Max",
                  "className": "title is-4",
                },
                "ref": null,
                "rendered": "iPhone XS Max",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Gold",
                  "className": "subtitle is-6",
                },
                "ref": null,
                "rendered": "Gold",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="card-image"
          >
            <figure
              className="image"
            >
              <img
                alt="iPhone XS Max"
                src="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
              />
            </figure>
          </div>,
          <div
            className="card-content"
          >
            <div
              className="media"
            >
              <div
                className="media-content"
              >
                <p
                  className="title is-4"
                >
                  iPhone XS Max
                </p>
                <p
                  className="subtitle is-6"
                >
                  Gold
                </p>
              </div>
            </div>
          </div>,
        ],
        "className": "card",
        "to": "/detail/01",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <figure
              className="image"
            >
              <img
                alt="iPhone XS Max"
                src="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
              />
            </figure>,
            "className": "card-image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="iPhone XS Max"
                src="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
              />,
              "className": "image",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "iPhone XS Max",
                "src": "https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "figure",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="media"
            >
              <div
                className="media-content"
              >
                <p
                  className="title is-4"
                >
                  iPhone XS Max
                </p>
                <p
                  className="subtitle is-6"
                >
                  Gold
                </p>
              </div>
            </div>,
            "className": "card-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="media-content"
              >
                <p
                  className="title is-4"
                >
                  iPhone XS Max
                </p>
                <p
                  className="subtitle is-6"
                >
                  Gold
                </p>
              </div>,
              "className": "media",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="title is-4"
                  >
                    iPhone XS Max
                  </p>,
                  <p
                    className="subtitle is-6"
                  >
                    Gold
                  </p>,
                ],
                "className": "media-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "iPhone XS Max",
                    "className": "title is-4",
                  },
                  "ref": null,
                  "rendered": "iPhone XS Max",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Gold",
                    "className": "subtitle is-6",
                  },
                  "ref": null,
                  "rendered": "Gold",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
